### PR TITLE
Add annotation to disable request ration warnings per namespace

### PR DESCRIPTION
**Pull request by glrf**
## Summary

Introduces an annotation to disable the request ratio webhook on a per namespace basis.

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.

### CODE REVIEW

File: webhooks/ratio_validator.go [hunk 1]
```diff
@@ -91,6 +91,46 @@ func (v *RatioValidator) Handle(ctx context.Context, req admission.Request) admi
 	return admission.Allowed("ok")
 }
 
+func (v *RatioValidator) recodObject(ctx context.Context, r *Ratio, req admission.Request) (*Ratio, error) {
```
Comment (bastjan) [inline webhooks/ratio_validator.go:94]: Record? 😅

File: webhooks/ratio_validator.go [hunk 2]
```diff
@@ -91,6 +91,46 @@ func (v *RatioValidator) Handle(ctx context.Context, req admission.Request) admi
...
+	err := v.client.Get(ctx, client.ObjectKey{
+		Name: nsName,
+	}, &ns)
+	if err != nil {
+		return false, err
+	}
+
+	disabled, ok := ns.Annotations[RatioValidatiorDisableAnnotation]
+	if !ok {
+		return false, err
```
Comment (bastjan) [inline webhooks/ratio_validator.go:129]: I prefer `nil` here so i don't have to back-reference what the variable means.

```suggestion
		return false, nil
```

File: webhooks/ratio_validator.go [hunk 3]
```diff
@@ -91,6 +91,46 @@ func (v *RatioValidator) Handle(ctx context.Context, req admission.Request) admi
...
+			return r, err
+		}
+		r = r.RecordStatefulSet(sts)
+	}
+	return r, nil
+}
+
+func (v *RatioValidator) isNamespaceDisabled(ctx context.Context, nsName string) (bool, error) {
+	ns := corev1.Namespace{}
+	err := v.client.Get(ctx, client.ObjectKey{
```
Comment (bastjan) [inline webhooks/ratio_validator.go:120]: Might also need RBAC permissions? 😊

**Review by bastjan — APPROVED** [review]

File: webhooks/ratio_validator.go (hunk 3 above)
Comment (glrf) [reply webhooks/ratio_validator.go:120]: Aah how did I forget that again :see_no_evil:

File: webhooks/ratio_validator.go (hunk 2 above)
Comment (glrf) [reply webhooks/ratio_validator.go:129]: Good point. I agree

File: webhooks/ratio_validator.go (hunk 1 above)
Comment (glrf) [reply webhooks/ratio_validator.go:94]: Yes :see_no_evil:

**Pull request merged**